### PR TITLE
fix(animations): always normalize style properties and values during compilation

### LINKED ISSUE #11582: animation state styles do not auto-suffix numbers with px for browsers
`state('name', style({ width: 500 })` should be rendered as `width: 500px` at the end of the animation.

### LINKED ISSUE #12481: Animations of border-color are not persisted on ios/safari
**I'm submitting a ...**  (check one with "x")

```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Animation of border-color starts but returns to previous state.

**Expected behavior**
Animation should be consistent on the rest of platforms, and apply the end state.

**Minimal reproduction of the problem with instructions**
http://plnkr.co/edit/n2Dw8g?p=preview

**Please tell us about your environment:**
IOS/Safari 10.0 Mobile/14A456 Safari/602.1
- **Angular version:** 2.1.1
- **Browser:** [iOS 10.0 Safari]
  
  
- **Language:** [ TypeScript 2]

### PR DESCRIPTION
**Pull request by matsko**
Closes #11582
Closes #12481

### CODE REVIEW

File: modules/@angular/compiler/src/schema/dom_element_schema_registry.ts [hunk 1]
```diff
@@ -373,4 +374,72 @@ export class DomElementSchemaRegistry extends ElementSchemaRegistry {
...
+
+  normalizeAnimationStyleProperty(propName: string): string {
+    return dashCaseToCamelCase(propName);
+  }
+
+  normalizeAnimationStyleValue(camelCaseProp: string, userProvidedProp: string, val: string|number):
+      string {
+    let unit: string = '';
+    var strVal = val.toString().trim();
+    if (_isPixelDimensionStyle(camelCaseProp) && val != 0 && val != '0') {
```
Comment (vicb) [inline modules/@angular/compiler/src/schema/dom_element_schema_registry.ts]: `val !== 0 && val !=='0'`

File: modules/@angular/compiler/src/schema/dom_element_schema_registry.ts [hunk 2]
```diff
@@ -373,4 +375,62 @@ export class DomElementSchemaRegistry extends ElementSchemaRegistry {
   }
 
   allKnownElementNames(): string[] { return Object.keys(this._schema); }
+
+  normalizeAnimationStyleProperty(propName: string): string {
```
Comment (tbosch) [inline modules/@angular/compiler/src/schema/dom_element_schema_registry.ts]: Please add unit tests for this method to https://github.com/angular/angular/blob/master/modules/@angular/compiler/test/schema/dom_element_schema_registry_spec.ts

File: modules/@angular/compiler/src/schema/dom_element_schema_registry.ts [hunk 3]
```diff
@@ -373,4 +375,62 @@ export class DomElementSchemaRegistry extends ElementSchemaRegistry {
...
+    return dashCaseToCamelCase(propName);
+  }
+
+  normalizeAnimationStyleValue(camelCaseProp: string, userProvidedProp: string, val: string|number):
```
Comment (tbosch) [inline modules/@angular/compiler/src/schema/dom_element_schema_registry.ts]: Please add unit tests for this method to https://github.com/angular/angular/blob/master/modules/@angular/compiler/test/schema/dom_element_schema_registry_spec.ts to check all the corner cases.

File: modules/@angular/compiler/src/schema/dom_element_schema_registry.ts [hunk 4]
```diff
@@ -373,4 +375,62 @@ export class DomElementSchemaRegistry extends ElementSchemaRegistry {
...
+      string {
+    let unit: string = '';
+    var strVal = val.toString().trim();
+    if (_isPixelDimensionStyle(camelCaseProp) && val !== 0 && val !== '0') {
+      if (typeof val === 'number') {
+        unit = 'px';
+      } else {
+        let valAndSuffixMatch = val.match(/^[+-]?[\d\.]+([a-z]*)$/);
+        if (valAndSuffixMatch && valAndSuffixMatch[1].length == 0) {
+          throw new Error('Please provide a CSS unit value for ' + userProvidedProp + ':' + val);
```
Comment (tbosch) [inline modules/@angular/compiler/src/schema/dom_element_schema_registry.ts]: I think there is no test for this error.

**Pull request merged**